### ISSUE TITLE
Contributing ruby bindings to Apache Arrow

### ISSUE THREAD

**Issue by wesm** [kou/arrow-glib#1]
hi @kou -- I wanted to see if you would like to start a `ruby/` subproject within the Apache Arrow codebase. We would love to have you involved (beyond the patches you've already provided), and to also set up continuous integration for Ruby within the Arrow repo

**Comment by kou** [comment]
It's interesting but Apache Arrow's development process (creating a issue on JIRA before implementing) is heavy for me... (I want to just implement without creating a issue.)

If CI is important for you, I can send a pull request to the Apache Arrow repository for Travis CI configuration that tests this project with the committed Apache Arrow.

**Comment by wesm** [comment]
You don't have to create an issue before you implement. It takes about 30 seconds to add a JIRA with a 1 line description of what you are doing. I open a lot of JIRAs because it personally helps me stay organized, but sometimes I only open one right when I'm submitting a patch that I've just finished. If you are collaborating with others, then it can help with communicating what you are doing.

Really JIRA is there in place of a CHANGELOG -- we can generate the summary of a release automatically from JIRA, e.g. https://github.com/apache/aurora/blob/master/CHANGELOG

**Comment by kou** [comment]
Does it mean that I follow the following cycle if I work on the Apache Arrow repository?

1. I implement a feature
2. I open an issue with one line description on JIRA
3. `git commit` with `ARROW-XXX: ...` commit message
4. `git push`

If it's right, I'm OK that I work on the Apache Arrow repository.

**Comment by wesm** [comment]
Yes -- you actually don't even need to include the JIRA number in your commit messages as long as it's in the title of your pull request.

**Comment by kou** [comment]
Thanks for confirming it.

I have one more question. I can't push git://git.apache.org/arrow.git directory. I need to create a pull request on https://github.com/apache/arrow like the following. Right?

1. `git checkout -b BRANCH_FOR_NEW_FEATURE`
2. `I implement a new feature
3. I open an issue with one line description on JIRA
4. `git commit` (I don't need to include `ARROW-XXX: ` into commit message)
5. `git push`
6. I open a pull request on GitHub for the new feature

(The steps are also OK to me.)

**Comment by wesm** [comment]
That's right

**Comment by kou** [comment]
OK. I'll send a pull request.
